docs/proposals/ha_compact.md

```diff
@@ -0,0 +1,88 @@
+# Thanos Compact - High Availability
+
+Status: **draft** | in-review | rejected | accepted | complete
+
+Implementation Owner: [@domgreen](https://github.com/domgreen)
+
+## Summary
+
+Thanos Compact currently runs as a singleton against a given bucket, this proposal outlines how we could create a highly avaliable
+setup so that given one compactor dies or is unable to reach a bucket another instance of `thanos compact` can take over the data compaction.
+
+## Motivation
+
+If a `thanos compact` instance dies we would like another to take over compaction of the data in a bucket.
+
+## Goals
+
+- ensure `thanos compact` can be run in a HA manner
+
+## Proposal
+
+An initial proposal of adding ETCD into `thanos compact` so that we can run master election against multiple components.
```

Comment:
What about the data? If the compactor dies after uploading a compacted block, but before deleting the old blocks could the new compactor fail because of overlapping blocks? Keeping the data about which blocks are compacted in etcd could help prevent that?

Reply:
Yes, extremely valid point, there might be lots of edge cases there like this. We might want to start putting some meta info about who produced the block (who compacted) etc.

We have: 
```
// Source is a real upload source of the block.
Source SourceType `json:"source"`
```
But we need some id of the exact thing for easier debug. Based on that info and note in etcd who is the peer and that it is down, we can safely delete the source blocks or something. Maybe we actually want to design it more here before merging? @domgreen  

We might want to depend on etcd even more largly in future - in case of object storage being completely eventual consistent.. but that's separate story. GCS and AWS are strongly consistent for our use cases, but not sure about other ones that are in plans or in PRs